### DISCUSSION TITLE
Licensing question

### DISCUSSION THREAD

**Discussion by ghost** [netblue30/firejail#6351] [Q&A]
Hi all. Recently I've been more actively present on IRC after a long time of not using it. Not limiting my presence to our #firejail channel on Libera, I noticed a shell script was being passed around that describes itself as `a bubblewrap wrapper that can convert/use firejail profiles on the fly`. As far as I'm aware the script's author hasn't distributed it anywhere (yet), nor is it available on GitHub, GitLab or the likes. Licensing questions started to bubble up left and right on the IRC channel in question. Not at all famailiar with the topic, I wonder if we should `add GPL-V2 license headers to all our profiles`?

Just asking, I don't know what's `proper procedure` in this context, nor if it's actually something that we should concern ourselves with.

**Comment by kmk3** [comment]
> Licensing questions started to bubble up left and right on the IRC channel in
> question. Not at all famailiar with the topic, I wonder if we should `add
> GPL-V2 license headers to all our profiles`?

That is a good question; I have wondered about it as well (and also for other
files).

Most source code files (and all man pages) say that they are licensed under the
GPL v2 or later in the copyright/license comment, so I'd assume that the
project as a whole is intended to be licensed under those terms.

There are also many files using the following license comment:

```
# License GPL v2
```

Which is ambiguous, as it can mean either `GPL-2.0-or-later` or `GPL-2.0-only`.

Given the context, presumably it means the former, so I'd replace those with:

```
# SPDX-License-Identifier: GPL-2.0-or-later
```

If that is also the intended license for the profiles, I'd add it to the first
line of each profile as well to clarify the licensing.

For reference, the following explains the SPDX part:

* <https://github.com/david-a-wheeler/spdx-tutorial>

Related:

* #5667

**Comment by ghost** [comment]
`GPL-2.0-or-later` is indeed how I interpret Firejail's licensing.

Thanks for the detailed response.
